### ISSUE TITLE
Do you plan to turn piles into CollectibleBehavior?

### ISSUE THREAD

**Issue by Craluminum2413** [FlyingSpaten/stonepiles#2]
Instead of completely destroying compatibility with mods that modify stones, I advise to use CollectibleBehavior.
Example of using CollectibleBehavior:
<https://github.com/Craluminum-Mods/ImmersiveCrafting/blob/main/src/CollectibleBehavior/BehaviorUseOnBucket.cs>

**Comment by FlyingSpaten** [comment]
Yes I do plan on addressing that problem. This is my first mod and well I did know from the start it was more of a hack. But thanks for the idea, I will try it out.

**Comment by FlyingSpaten** [comment]
By now I created a first draft for a new version which uses the CollectibleBehavior. But as far as I understand the VS Code, it will only work with the stone-item (or at least not with clay because stone calls the behavior directly?).
I will give it a bit more work thoug..

**Comment by FlyingSpaten** [comment]
Finally got to work on the project again. I'm sorry it takes this long, but sadly I couldn't find more time. I have implemented the CollectibleBehavior for the Stone-Piles only for now, because I could not get it to work with the other ones. But I plan on adding that in the future.

**Comment by Craluminum2413** [comment]
You can also look at [BlockEntityBehavior](https://github.com/anegostudios/vsapi/blob/master/Common/Collectible/Block/BlockEntityBehavior.cs)

**Comment by FlyingSpaten** [comment]
I dont know if or how the BlockEntityBehavior will help with the compatibility with other mods. How do you propose to use it?

**Comment by Craluminum2413** [comment]
No idea, but you can also look at how Ground Storage is implemented
<https://github.com/anegostudios/vssurvivalmod/blob/master/CollectibleBehavior/BehaviorGroundStorable.cs>
<https://github.com/anegostudios/vssurvivalmod/blob/master/Block/BlockGroundStorage.cs>
<https://github.com/anegostudios/vssurvivalmod/blob/master/BlockEntity/BlockEntityGroundStorage.cs>
